test/test_servers.py

```diff
@@ -0,0 +1,197 @@
+import pytest
+from tabulate import tabulate
+from click.testing import CliRunner
+
+from faculty_cli.cli import cli, _server_spec
+
+from faculty.clients.server import ServerClient
+from test.fixtures import (
+    PROJECT,
+    SHARED_SERVER,
+    DEDICATED_SERVER,
+    DEDICATED_RESOURCE,
+    SHARED_RESOURCE,
+    USER_ID,
+)
+
+
+@pytest.fixture
+def mock_update_check(mocker):
+    mocker.patch("faculty_cli.update.check_for_new_release")
+
+
+@pytest.fixture
+def mock_check_credentials(mocker):
+    mocker.patch("faculty_cli.cli._check_credentials")
+
+
+@pytest.fixture
+def mock_user_id(mocker):
+    mocker.patch("faculty_cli.auth.user_id", return_value=USER_ID)
+
+
+def test_list_all_servers(
+    mocker,
+    mock_update_check,
+    mock_check_credentials,
+    mock_profile,
+    mock_user_id,
+):
+    runner = CliRunner()
+    mocker.patch("faculty_cli.cli._list_projects", return_value=[PROJECT])
+    schema_mock = mocker.patch("faculty_cli.cli.ServerSchema")
+    mocker.patch.object(ServerClient, "_get", return_value=[DEDICATED_SERVER])
+    result = runner.invoke(cli, ["server", "list"])
+    assert result.exit_code == 0
+    assert (
+        result.output
+        == tabulate(
+            [(PROJECT.name, DEDICATED_SERVER.name)],
+            ("Project Name", "Server Name"),
+            tablefmt="plain",
+        )
+        + "\n"
+    )
+    ServerClient._get.assert_called_once_with(
+        "/user/{}/instances".format(USER_ID), schema_mock.return_value
+    )
+
+
+def test_list_all_servers_verbose(
+    mocker,
+    mock_update_check,
+    mock_check_credentials,
+    mock_profile,
+    mock_user_id,
+):
+    runner = CliRunner()
+    schema_mock = mocker.patch("faculty_cli.cli.ServerSchema")
+    mocker.patch("faculty_cli.cli._list_projects", return_value=[PROJECT])
+    mocker.patch.object(ServerClient, "_get", return_value=[DEDICATED_SERVER])
+    result = runner.invoke(cli, ["server", "list", "--verbose"])
+    assert result.exit_code == 0
+    assert (
+        result.output
+        == tabulate(
```

Comment:
It would be more idiomatic to mock `tabulate` rather than call it twice (once in the actual code and once in the tests) and compare the output.